docs/v3/examples/atproto-dashboard-with-prefect-assets.mdx

```diff
@@ -0,0 +1,256 @@
+    """
+    print("Fetching raw data...")
+
+    # In the real implementation, this fetches from ATProto APIs
+    # See: https://github.com/zzstoatzz/atproto-dashboard/blob/main/src/atproto_dashboard/assets.py
+    data = {
+        "items": ["item1", "item2", "item3"],
+        "fetched_at": datetime.now(timezone.utc).isoformat(),
+        "count": 3,
+    }
+
+    print(f"✓ Fetched {data['count']} items")
+    return data
+
+
+@materialize(processed_data_asset)
+def process_data(raw_data: dict[str, Any]) -> dict[str, Any]:
+    """
+    Process raw data into a structured format.
+
+    This demonstrates automatic dependency tracking - by accepting raw_data as a
+    parameter, Prefect knows this asset depends on raw_data_asset and will ensure
+    it's materialized first.
+
+    Args:
+        raw_data: Data from the raw_data_asset (automatically injected by Prefect)
+
+    Returns:
+        Processed data ready for analytics
+    """
+    print(f"Processing {raw_data['count']} items...")
+
+    # In the real implementation, this stores data to S3 with partitioning
+    # See: https://github.com/zzstoatzz/atproto-dashboard/blob/main/src/atproto_dashboard/assets.py
+    processed = {
+        "items": [item.upper() for item in raw_data["items"]],
+        "processed_at": datetime.now(timezone.utc).isoformat(),
+        "source_count": raw_data["count"],
+    }
+
+    # Optionally persist to local storage (in prod, this would be S3)
+    storage_dir = Path("./data")
+    storage_dir.mkdir(exist_ok=True)
+    with open(storage_dir / "processed.json", "w") as f:
+        json.dump(processed, f, indent=2)
+
+    print(f"✓ Processed and stored {len(processed['items'])} items")
+    return processed
+
+
+@materialize(analytics_asset)
+def create_analytics(processed_data: dict[str, Any]) -> dict[str, Any]:
+    """
+    Generate analytics from processed data.
+
+    This demonstrates:
+    1. Chained dependencies (depends on processed_data, which depends on raw_data)
+    2. Artifact creation for observability in the Prefect UI
+
+    Args:
+        processed_data: Data from the processed_data_asset
+
+    Returns:
+        Analytics summary
+    """
+    print("Creating analytics...")
+
+    # In the real implementation, this runs dbt transformations
+    # See: https://github.com/zzstoatzz/atproto-dashboard/blob/main/src/atproto_dashboard/dbt_flow.py
+    analytics = {
+        "total_items": len(processed_data["items"]),
+        "source_timestamp": processed_data["processed_at"],
+        "created_at": datetime.now(timezone.utc).isoformat(),
+    }
+
+    # Create a Prefect artifact for observability
+    create_markdown_artifact(
+        key="analytics-summary",
+        markdown=f"""# Analytics Summary
+
+- **Total Items**: {analytics["total_items"]}
+- **Created**: {analytics["created_at"]}
+- **Source**: {analytics["source_timestamp"]}
+
+This artifact appears in the Prefect UI for observability.
+        """,
```

Comment:
I like using `textwrap.dedent` for multiline strings like this to keep the formatting nice.